### DISCUSSION TITLE
How to detect multiple tables in document image and convert into data frames

### DISCUSSION THREAD

**Discussion by PoornaSaiNagendra** [mindee/doctr#622] [Q&A]
I want to store the text inside a table in a data frame but the output from doctr has no information about the position of the table starts and ends. In order to create a data frame, I require the number of tables present in a document image, number of rows and columns(structure of table) present. Also, there are a few other complexities like column head contains more than one sub-columns, borderless table, semi border table. Could you help me resolve the table structure? Thanks for any help you can provide.

**Comment by charlesmindee** [comment]
Hi @PoornaSaiNagendra, 

Thanks for your interest in doctr!
For the moment we do not provide table detection/information extraction in doctr, we are working on it.
In the meantime, you can try to do it by yourself using the box coordinates: you can search for vertically align boxes and horizontally align boxes to reconstruct lines and columns for instance.

I hope this answer your question!

**Comment by felixdittrich92** [comment]
@PoornaSaiNagendra 
I have currently a model to segment the table rows and columns in a document and to extract the information as dataframe i can provide it if you want to test it !? :) But note this is currently not optimal it takes ~4sec on CPU without OCR (In a later point i will investigate again and than i think we will also find a way into doctr :) )

**Reply by PoornaSaiNagendra** [reply]
Thank you for your great help. It could be beneficial for us if you could share it.

**Reply by PoornaSaiNagendra** [reply]
@felixdittrich92 
I have tried the TableNet project that you suggested in #524 on my data. 

Here I am providing the image for your reference.
![3](https://user-images.githubusercontent.com/42320447/141927847-866693e6-c45a-4eaf-a03f-bcd85b2b7afb.png)
Image source: Google Images

And I am getting this as output:

<class 'pandas.core.frame.DataFrame'>
   0
0  

Even I tried on bordered table images as well but I got similar kind of results.
.

**Reply by felixdittrich92** [reply]
@PoornaSaiNagendra 
Oh yes i see this will not work the model was trained on marmot dataset (which contains only perfect scanned papers) ..
you can take a look at [this](https://github.com/DevashishPrasad/CascadeTabNet) but i think to detect borderless tables in mobilephone images will be also for this tricky :sweat_smile:

**Reply by PoornaSaiNagendra** [reply]
@felixdittrich92 
Yeah I even tried [this](https://github.com/DevashishPrasad/CascadeTabNet)  as well. It is doing good on bordered tables but while changing it to cpu it is throwing error as it is using mmdectection as backbone which using uses cuda😅.
